torch/csrc/jit/register_special_ops.cpp

```diff
@@ -64,32 +64,16 @@ at::Tensor castTensorTo(
   return self;
 }
 
-int64_t list_size(const IValue& list) {
-  if (list.isGenericList()) {
-    return list.toGenericListRef().size();
-  } else if (list.isIntList()) {
-    return list.toIntListRef().size();
-  } else if (list.isDoubleList()) {
-    return list.toDoubleListRef().size();
-  } else if (list.isBoolList()) {
-    return list.toBoolList().size();
-  }
-  AT_ASSERTM(0, "Unexpected list type", list);
-}
-
 std::vector<int64_t> compute_sizes(const IValue& seq) {
   std::vector<int64_t> sizes;
-  // because bool, int, and float lists are specialized, inner array will
-  // will not be generic list
-  auto seq_recur = seq;
-  while (seq_recur.isGenericList()) {
-    auto seq_list = seq_recur.toGenericListRef();
-    auto length = seq_list.size();
-    AT_ASSERT(length != 0);
-    sizes.push_back(length);
-    seq_recur = seq_list[0];
+  auto seq_recur = seq.toGenericList();
+  while (true) {
+    sizes.push_back(seq_recur.size());
+    if (seq_recur.size() == 0 || !seq_recur.get(0).isGenericList()) {
```

Comment:
side note: maybe we should rename `isGenericList` to `isList` because all lists are generic now.

Reply:
Yes, I will put in the renaming PR. I might leave isGenericList in as a deprecated name depending on how many places in fbcode use isGenericList